Frankenstein

comma opened the pull request:
"Meat limbs now can be attached. Use limb on missing area, then hemostat to finalize it."
"Limbs from other races can be now attached. They'll cause rejection, but it can be kept at bay with spaceacilline to some point. Species special attack is carried over too, i.e. you can clawn people if you sew a cathand to yourself."
"Limbs that are left in open will rot in ~7 minutes. Use freezers or cryobags to stop it. You can still attach them, but you wish you couldn't."

EDIT: Now in pictures!
![comics!](http://i.gyazo.com/b8f6599032e0e2bb4e893157572ed048.png)

comma committed:
Generalized head reattachment to work with all limbs.
Now you can attach your useless chunks of meat back!
Needs second step to make meat limb usable, use hemostat/cable/mousetrap.

Changes how organs 'die'. Instead of damage they now acquire germs, 2-6 points per tick, twice faster after hitting acute infection stage. Once they reach necrosis stage, they die. All in all it takes ~7 minutes for organ to go bad without a freezer.
As a bonus, you can now attach robogroins and roboheads (no brain included)
Fixes stumps being un-cuttable, which prevented all kinds of fun stuff with replacements.
Removed ORGAN_DESTROYED check from processing, take_damage should be taking care of that now.

comma committed:
Adds a portable organs/lunch freezer

comma committed:
Frankenstein edition:
Limbs now store their own dna/species info. This allows transplanted limbs show up as they are, instead of conforming to mob. Vox-cat-monkey hybrids, rainbow skrells, go wild.
Also changed rejection to cause infection instead of straight up organ damage - wouldn't want your newly acquired catpaws fracture out of blue.
As a bonus, attacks are tracked per limb, so for example having kitty claws will give you the corresponding attack

comma committed:
Changelog

comma committed:
Merge branch 'dev' of https://github.com/Baystation12/Baystation12 into frankenstein

PsiOmegaDelta commented:
Seems like a future PR could be to make it possible to make bodyparts less likely to be rejected by using genetics.

comma commented:
Hehehe genetics. But yeah, either that or some kind of chem that goes over rejecting organs and sets their rejecting to -1 (and a check that stops rejection for such limbs)

Iamgoofball commented:
If you're gonna do rotting limbs and injecting chems into, why not port formaldehyde as embalming fluid for limbs? It could prevent or delay the rot. This is pretty cool BTW, :+1:

Zuhayr commented on code/modules/surgery/headreattach.dm:
```diff
@@ -1,33 +1,73 @@
-		user.visible_message("\red [user]'s hand slips, damaging [target]'s neck!", \
-		"\red Your hand slips, damaging [target]'s neck!")
+		var/obj/item/organ/external/E = tool
+		user.visible_message("\red [user]'s hand slips, damaging [target]'s [E.amputation_point]!", \
```
Span classes!

comma replied:
But all other surgeries do that :(

Zuhayr commented:
This is exciting. Glad you did this so I don't need to.

comma committed:
Fixed changelog
Added some info about not quite attached limbs to examination/scanner.
Added rejection info to advanced scanner
Span classes (gotta start somewhere I guess)
Also zombie goast file pls go

GinjaNinja32 commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -628,3 +628,15 @@
+
+/obj/item/weapon/storage/box/freezer
+	name = "portable freezer"
+	desc = "This nifty schok-resistant device will keep your 'groceries' nice and non-spoiled."
```
shock?

GinjaNinja32 commented on code/modules/surgery/limb_reattach.dm:
```diff
@@ -0,0 +1,129 @@
+	allowed_tools = list(
+	/obj/item/weapon/hemostat = 100,	\
+	/obj/item/stack/cable_coil = 75, 	\
+	/obj/item/device/assembly/mousetrap = 20
```
I... what...

Iamgoofball replied:
Ghetto surgery I guess?

GinjaNinja32 replied:
With a mousetrap?

comma replied:
Yeeep, that's fairly consistent replacement for hemostat in other procedures. They kinda look alike

Iamgoofball replied:
i know the reasoning is there but it is slightly beyond the grasp of my mins at 12 am at night

I'll sleep on it and ponder mousetrap ghetto surgery in the mean time

GinjaNinja32 commented on code/modules/organs/organ.dm:
```diff
@@ -86,6 +98,11 @@ var/list/organ_cache = list()
+/obj/item/organ/examine(mob/user)
+	..(user)
+	if(status & ORGAN_DEAD)
+		user << "<spanc class='notice'>The decay has set in.</span>"
```
`spanc`

comma committed:
tyops fiexs

Iamgoofball commented:
Does this mean I can rip apart an IPC and staple rotting limbs onto it?

comma commented:
Not sure if IPCs were updated to current organ code, but I guess yeah.

Iamgoofball commented on code/modules/organs/organ.dm:
```diff
@@ -114,24 +131,23 @@ var/list/organ_cache = list()
-			rejecting = 1
+	if(dna)
+		if(!rejecting)
+			if(blood_incompatible(dna.b_type, owner.dna.b_type, species, owner.species))
```
Might wanna mention that organ blood types matter too, not just the species type

Zuhayr replied:
blood_incompatible() checks both.

comma replied:
>dna.b_type
>owner.dna.b_type
b_type - blood type

scrdest commented:
FUND HE : :heavy_multiplication_x: :100:

Techhead0 commented:
Would it be weird to ask for finalizing the surgery to require a hemostat plus cautery instead of just a hemostat?
I think you would just need to change /limb/attach/end_step to include `E.open = 1` and /limb/connect/can_use to include a `&& E.open == 1` in the return.

comma commented:
Dunno, I think limbs attachment would take long enough as it is. You also need to cut off the stump first too (like for robolimbs), so it's actually 3 steps.

comma commented:
;-; too late to the dev train

DeeUnderscore commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -628,3 +628,15 @@
+	max_w_class = 3
+	can_hold = list(/obj/item/organ, /obj/item/weapon/reagent_containers/food, /obj/item/weapon/reagent_containers/glass)
+	max_storage_space = 21
+	use_to_pickup = 1 // for picking up broken bulbs, not that most people will try
```
That comment made me do a double take

comma replied:
Copypaste :V

DeeUnderscore commented:
Do we have a way of figuring out the DNA/species from a severed part? There may be situations where you need to figure out what body part belongs to whom without using a grinder.

Actually, wait, are we even supposed to be able to grind hands in the chem lab grinder? Cause we can.

comma commented:
I didn't add any, and why not? Sounds reasonable!

Ccomp5950 commented:
:-1:   Does not remove all aliums

mwerezak commented:
@DeeUnderscore @comma Is it possible to remove a blood sample from a severed part and run it through the spectrometer?

comma commented:
I don't think so, since it's not reagent container and syringe won't be able to pick up it. You can just wave it around until it makes a puddle on the flor and then use detective scanner

mwerezak commented:
Right, it's only the internal organs that are reagent containers.

comma commented:
Only when bitten once :V

mwerezak commented:
Right, forgot that.

Draconomial commented:
Question: Hulk limbs?

Looking forward to seeing more diverse surgery/arrests in xenobiology.

On Thu, May 21, 2015 at 6:05 PM, mwerezak <notifications@github.com> wrote:

> Right, forgot that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Baystation12/Baystation12/pull/9370#issuecomment-104465782>
> .
>

comma commented:
Anyone?

Zuhayr commented:
Fuck the police.
